hudi-spark-datasource/hudi-spark/pom.xml

```diff
@@ -364,6 +395,12 @@
      <scope>provided</scope>
     </dependency>
 
+    <dependency>
+      <groupId>org.apache.hive</groupId>
+      <artifactId>hive-storage-api</artifactId>
+      <version>2.7.2</version>
+    </dependency>
+
```

Comment:
this means compile scope for hudi-spark. how does this affect the bundle jar? do we shade it? want to understand the risk of setting a special dependency here